### ISSUE TITLE
Decoding of HTML entities (ie. &#039;) does not work.

### ISSUE THREAD

**Issue by michaelrsweet** [michaelrsweet/mxml#8]
Version: 2.0
Original reporter: Michael Sweet

I found a function in the file mxml-file.c called mxml_get_entity. 
This has a few lines that read as follows;

  if (entity[1] == '#')
  {
    if (entity[2] == 'x')
      ch = strtol(entity + 3, NULL, 16);
    else
      ch = strtol(entity + 2, NULL, 10);
  }

It seems the index into "entity" is actually off by one.  I changed the
code to read as follows;

  if (entity[0] == '#')
  {
    if ((entity[1] == 'x') || (entity[1] == 'X'))
      ch = strtol(entity + 2, NULL, 16);
    else
      ch = strtol(entity + 1, NULL, 10);
  }

Basically, decremented the index into "entity" by one since I found the
# sign was actually at the start of "entity".  I also just added a
quick check for an uppercase "X".
